nixos/modules/services/misc/gitlab.nix

```diff
@@ -655,36 +655,43 @@ in {
     # here.
     systemd.services.gitlab-postgresql = let pgsql = config.services.postgresql; in mkIf databaseActuallyCreateLocally {
       after = [ "postgresql.service" ];
+      bindsTo = [ "postgresql.service" ];
       wantedBy = [ "multi-user.target" ];
-      path = [ pgsql.package ];
+      path = [
+        pgsql.package
+        pkgs.util-linux
+      ];
       script = ''
         set -eu
 
-        PSQL="${pkgs.util-linux}/bin/runuser -u ${pgsql.superUser} -- psql --port=${toString pgsql.port}"
+        PSQL() {
+            runuser -u ${pgsql.superUser} -- psql --port=${toString pgsql.port} "$@"
```

Comment:
This doesn't need to run as `root` anymore. Can set `User` to `pgsql.superUser` and should be fine.

Reply:
Good point! It's fixed now.